build-data-capturing-gradle-samples/capture-thermal-throttling/gradle-thermal-throttling.gradle.kts

```diff
@@ -0,0 +1,142 @@
+import org.gradle.api.internal.project.ProjectInternal
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters
+import com.gradle.enterprise.gradleplugin.GradleEnterpriseExtension
+import org.gradle.internal.os.OperatingSystem
+import com.gradle.scan.plugin.BuildScanExtension
+
+import java.nio.charset.Charset
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.Executors
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.TimeUnit
+import java.util.function.Function
+import java.util.Queue
+
+import org.gradle.tooling.events.OperationCompletionListener
+import org.gradle.tooling.events.FinishEvent
+
+/**
+ * This Gradle script captures the thermal throttling level and adds it as a tag.
+ * Some parameters can be tweaked according to your build:
+ * - SAMPLING_INTERVAL_IN_SECONDS, frequency at which the capture command is run
+ * - THROTLLING_LEVEL, list of throttling levels and value ranges (to be compared with captured throttling average value)
+ *
+ * WARNINGS:
+ * - This is supported on MacOS only.
+ */
+
+project.extensions.configure<GradleEnterpriseExtension>() {
+    buildScan {
+        val api = buildScan
+        if (OperatingSystem.current().isMacOsX()) {
+            // Register Thermal throttling service
+            val throttlingServiceProvider =
+                gradle.sharedServices.registerIfAbsent("thermalThrottling", ThermalThrottlingService::class.java) {}
+            // start service
```

Comment:
Let's add a comment here that this BuildEventsListenerRegistry is necessary in order to mark this BuildService as configuration cache compatible.

Reply:
How about ?

>   // Service bootstrapping compatible with the configuration cache

Reply:
How about this:

> // We register a BuildEventsListenerRegistry in order to mark the BuildService as configuration cache compatible but is otherwise unused.